### PR TITLE
Justin krs/customer save changes

### PR DESCRIPTION
**Pull request by justinengle-krs**
Hookup the API calls on customer save changes

- Currently using alerts and confirms until the designer is back to confirm the designs
- Confirm the handling of stale data by opening 2 tabs, edit on one, and then save. Go back to first tab and see that the confirmation pops up when trying to save.
-TODO: Grey out save changes if no changes have been made, added new trello card for this - some quirk is preventing this reactive from working properly.

### CODE REVIEW

File: src/components/Customer/CustomerDetail.svelte [hunk 1]
```diff
@@ -56,37 +71,100 @@
...
+    if (systemChannel.type !== 'trade') {
+      alert('Customer saved');
+      loading = false;
+      invalidateCache();
+      return;
     }
+
+    await saveCustomerUser();
+
+    invalidateCache();
```
Comment (TheSuperRealJP) [inline src/components/Customer/CustomerDetail.svelte:157]: There a bit of duplication here for `loading = false;` and `invalidateCache();` at lines 150 - 151 and 157 - 159. Any way we can reorder some things so we only have these lines once?

File: src/lib/api/api.ts [hunk 2]
```diff
@@ -33,6 +33,7 @@ async function handleResponse(response: Response): Promise<any> {
     const data = await response.json();
 
     if (!response.ok) {
+      console.log(response);
```
Comment (TheSuperRealJP) [inline src/lib/api/api.ts:36]: Temporary log, or?

Comment (justinengle-krs) [reply src/lib/api/api.ts]: Mistake was commit on wrong branch

File: src/components/Customer/CustomerDetail.svelte (hunk 1 above)
Comment (justinengle-krs) [reply src/components/Customer/CustomerDetail.svelte:153]: Sorted

**Review by TheSuperRealJP — APPROVED** [review]

File: src/stories/routes/Customer.stories.svelte [hunk 3]
```diff
@@ -8,6 +8,8 @@
   import MockCustomerUsersRepo from '../test-data/customer-users/mock-customer-user-repo';
```
Comment (matthampton-stock2shop) [inline src/stories/routes/Customer.stories.svelte:8]: I did a chromatic build and there is some content that has disappeared - not sure if it is meant to be that way?

https://www.chromatic.com/test?appId=63ef6cddab1dec384359e1cd&id=64107c0a9685739f27edc95c

Comment (justinengle-krs) [reply src/stories/routes/Customer.stories.svelte:8]: My mistake on resolving merge conflicts. Running new build now

Comment (justinengle-krs) [reply src/stories/routes/Customer.stories.svelte:8]: https://www.chromatic.com/build?appId=63ef6cddab1dec384359e1cd&number=18

**Comment by TheSuperRealJP** [comment]
Is it fine if this isn't a real email address?
image

**Comment by justinengle-krs** [comment]
> 

@TheSuperRealJP I do have a card to confirm some form validation rules. Where these kinds of things will be implemented.
https://trello.com/c/kHWDsYVj/26-confirm-validation-rules

**Review by TheSuperRealJP — DISMISSED** [review]

File: src/components/Customer/CustomerDetail.svelte [hunk 4]
```diff
@@ -100,8 +171,8 @@
   <div>
     <PageHeader title="{customerVersionControlled.workingCopy.first_name} {customerVersionControlled.workingCopy.last_name}" hasBackButton={true}>
       <div class="flex items-center gap-6">
-        <IsActiveDropdown bind:isActive={customerUserVersionControlled.workingCopy.active} />
-        <Btn onClick={saveChanges} size="large" variant="solid">Save changes</Btn>
+        <IsActiveDropdown bind:isActive={customerVersionControlled.workingCopy.active} />
+        <Btn onClick={saveChanges} bind:disabled={changesMade} size="large" variant="solid">Save changes</Btn>
```
Comment (matthampton-stock2shop) [inline src/components/Customer/CustomerDetail.svelte:175]: I don't see where `changesMade` gets modified.. Is that still a TODO?

File: src/repos/users-repo.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { get } from '@/lib/api/api';
+import { get, put } from '@/lib/api/api';
```
Comment (matthampton-stock2shop) [inline src/repos/users-repo.ts:1]: Is this used?

File: src/components/Customer/CustomerDetail.svelte (hunk 4 above)
Comment (justinengle-krs) [reply src/components/Customer/CustomerDetail.svelte:175]: Yes - I have it as a task on the trello. Wasn't updating as expected so will get back to that

File: src/repos/users-repo.ts (hunk 5 above)
Comment (justinengle-krs) [reply src/repos/users-repo.ts]: No - resolved

**Review by matthampton-stock2shop — APPROVED** [review]

**Pull request merged**